float32 rpm
bool spin_direction